app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -418,6 +418,12 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             components.analytics.experiments.apply {
                 initialize()
                 setupNimbusObserver(this)
+                // To workaround a caching bug in Nimbus that appears when we try to query an
+                // experiment **before** `FxNimbus.initialize` is called we need to set the `api`
+                // value again so that we can still access the NimbusApi that is wrapped
+                // in `FxNimbus.initialize.getSdk`.
+                // See: https://github.com/mozilla-mobile/fenix/issues/26041
+                FxNimbus.api = this
```

Comment:
> This is fixed [mozilla/application-services#4999](https://github.com/mozilla/application-services/pull/4999) first released in [v93.6.0](https://github.com/mozilla/application-services/releases/tag/v93.6.0)
> 
> It should already be in `main`, and `releases_v104.0.0`.
> 
> Added [mozilla-mobile/android-components#12567](https://github.com/mozilla-mobile/android-components/pull/12567) to land on `main`, this should be uplifted to `releases_v104.0.0`.

@jhugman Upgrading to the latest version (that is in https://github.com/mozilla-mobile/android-components/pull/12592), I'm still not able to get the experiment to work without adding this line.

At this point, I'm more inclined to leaving this line in here and filing a bug to have it fixed.

Reply:
Filed https://github.com/mozilla/application-services/issues/5075

Reply:
@jonalmeida @jhugman I don't think we can land the `FxNimbus.api = this` assignment because it's not thread-safe. It's a `public var` without any visibility guarantees, i.e., it's not volatile or synchronized, and also executed on an IO thread here.

Maybe that's also part of the problem? I know @jhugman made `FeatureHolder` thread-safe but this assignment isn't (to be fair I'd think it's also not meant to be called like this.)

Reply:
>  // To workaround a caching bug in Nimbus that appears when we try to query an experiment **before** `FxNimbus.initialize` ...

I don't understand how this can happen? `FxNimbus.initialize` is called on the main thread as part of [setupMegazord](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L405) before we launch on IO, and `setupMegazord` is called before we initialize the [engine](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L180-L188).

What is querying an experiment before `setupMegazord` (and therefore `FxNimbus.initialize`) are called? Or is the problem that something queries an experiment while Nimbus is initializing (not the `FxNimbus.initialize` call itself)?

Reply:
> to be fair I'd think it's also not meant to be called like this

This is correct. This is a workaround, using the old API, which we've deprecated, and are trying to remove.

As discussed, the fix is to upgrade the FML tooling to at least `v93.6.0` (currently on `v93.5.0`).

That said, there is an issue with upgrading to `latest` (currently `v93.8.0`) due to https://github.com/mozilla/application-services/issues/5079 .

Reply:
@jhugman Above @jonalmeida mentioned that upgrading to the latest version **didn't** fix the problem. We should take another look together, please.